Assets/Scripts/GooglePlayDownloaderImpl.cs

```diff
@@ -0,0 +1,121 @@
+using UnityEngine;
+using System.IO;
+using System;
+
+internal class GooglePlayObbDownloader : IGooglePlayObbDownloader
+{
+    private static AndroidJavaClass EnvironmentClass = new AndroidJavaClass("android.os.Environment");
+    private const string Environment_MediaMounted = "mounted";
+
+    public string PublicKey { get; set; }
+
+    private void ApplyPublicKey()
+    {
+        if (string.IsNullOrEmpty(PublicKey))
+        {
+            Debug.LogError("GooglePlayObbDownloader: The public key is not set - did you forget to set it in the script?\n");
+        }
+        using (var downloaderServiceClass = new AndroidJavaClass("com.unity3d.plugin.downloader.UnityDownloaderService"))
+        {
+            downloaderServiceClass.SetStatic("BASE64_PUBLIC_KEY", PublicKey);
+            // Used by the preference obfuscator
+            downloaderServiceClass.SetStatic("SALT", new byte[] { 1, 43, 256 - 12, 256 - 1, 54, 98, 256 - 100, 256 - 12, 43, 2, 256 - 8, 256 - 4, 9, 5, 256 - 106, 256 - 108, 256 - 33, 45, 256 - 1, 84 });
+        }
+    }
+
+    public void FetchOBB()
+    {
+        ApplyPublicKey();
+        using (var unityPlayerClass = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
+        {
+            var currentActivity = unityPlayerClass.GetStatic<AndroidJavaObject>("currentActivity");
+            var intent = new AndroidJavaObject("android.content.Intent",
+                                                currentActivity,
+                                                new AndroidJavaClass("com.unity3d.plugin.downloader.UnityDownloaderActivity"));
+
+            const int Intent_FLAG_ACTIVITY_NO_ANIMATION = 0x10000;
+            intent.Call<AndroidJavaObject>("addFlags", Intent_FLAG_ACTIVITY_NO_ANIMATION);
+            intent.Call<AndroidJavaObject>("putExtra", "unityplayer.Activity",
+                                                        currentActivity.Call<AndroidJavaObject>("getClass").Call<string>("getName"));
+            try
+            {
+                currentActivity.Call("startActivity", intent);
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError("GooglePlayObbDownloader: Exception occurred while attempting to start DownloaderActivity - is the AndroidManifest.xml incorrect?\n" + ex.Message);
+            }
+        }
+    }
+
+    public string GetExpansionFilePath()
+    {
+        if (EnvironmentClass.CallStatic<string>("getExternalStorageState") != Environment_MediaMounted)
+            return null;
+
+        const string obbPath = "Android/obb";
+        using (var externalStorageDirectory = EnvironmentClass.CallStatic<AndroidJavaObject>("getExternalStorageDirectory"))
+        {
+            var externalRoot = externalStorageDirectory.Call<string>("getPath");
+            return string.Format("{0}/{1}/{2}", externalRoot, obbPath, ObbPackage);
```

Comment:
Could we cache this?

Reply:
Yes. I was thinking the path can actually change in between calls (like, SD card ejected). However, it will be handled in line 53; whatever is returned here should not change. Will fix.